### DISCUSSION TITLE
prevent caching some pages with ssg

### DISCUSSION THREAD

**Discussion by milad-afkhami** [vercel/next.js#29378] [Help]
Hi
I'm using next.js v11.0.1
I'm currently developing a music app.
this app has a playlist page ( pages/playlist/[slug].js )
I have two problems:
1. playlists can be public or private
users can see their playlist but no user can see other user's private playlists.
I want to generate public playlists statically
but for playlists that didn't generate at build time ( e.g. private playlists ), there is a case when a user comes to see their playlist and then next.js caches their response and shows that to other users(which should not happen because that playlist meant to be private).
2. the second problem is that a playlist can get public or private at any time so how can I remove a playlist that was public and so popular that I generated it at build time but now got private or the owner deleted it.
for preventing these two problems I am currently showing skeleton loading at first load and then call the API to get the response.
can anyone tell me how can I solve these two problems?

**Comment by leerob** [comment]
Do the public pages need to be indexed for SEO? My assumption is yes.

- You can have public pages which need to be pre-rendered use `getStaticProps` with `revalidate` to update content
- For private pages, you can pre-render a loading state, followed by [fetching on the client side](https://swr.vercel.app/docs/with-nextjs)

**Reply by milad-afkhami** [reply]
yes SEO does matter
but two more things:

- in the second step of your solution, which `fallback` value( getStaticPaths )  should I use? probably `fallback: true` and on the client-side making use of `router.isFallback`? 
- If I wanted to use SSR, which `fallback` value should I use to tell next.js **not** to cache this page because it belongs only to this specific user and should not get cached and served for the others? can I use `notFound` property?

**Reply by leerob** [reply]
You don't need `fallback` in the second step. Just use a React book, like SWR or `useEffect`, and fetch the data on the client-side.

If you want SSR, you also don't need `fallback` - you want to use `getServerSideProps`.
